library/core/functions.general.php

```diff
@@ -2329,37 +2329,11 @@ function joinRecords(&$data, $column = '', $unset = false, $checkCategoryPermiss
      * @param mixed $value
      * @param int|null $options
      * @return string
-     * @throws Exception
+     * @throws Exception If an error occurred while encoding.
+     * @deprecated 2.8 Use \Vanilla\Utility\StringUtils::jsonEncodeChecked instead.
      */
     function jsonEncodeChecked($value, $options = null) {
-         if ($options === null) {
-            $options = JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES;
-        }
-        $encoded = json_encode($value, $options);
-        $errorMessage = null;
-        switch (json_last_error()) {
-            case JSON_ERROR_NONE:
-                // Do absolutely nothing since all went well!
-                break;
-            case JSON_ERROR_UTF8:
-                $errorMessage = 'Malformed UTF-8 characters, possibly incorrectly encoded';
-                break;
-            case JSON_ERROR_RECURSION:
-                $errorMessage = 'One or more recursive references in the value to be encoded.';
-                break;
-            case JSON_ERROR_INF_OR_NAN:
-                $errorMessage = 'One or more NAN or INF values in the value to be encoded';
-                break;
-            case JSON_ERROR_UNSUPPORTED_TYPE:
-                $errorMessage = 'A value of a type that cannot be encoded was given.';
-                break;
-            default:
-                $errorMessage = 'An unknown error has occurred.';
-        }
-        if ($errorMessage !== null) {
-            throw new Exception("JSON encoding error: {$errorMessage}", 500);
-        }
-        return $encoded;
+        return \Vanilla\Utility\StringUtils::jsonEncodeChecked($value, $options);
```

Comment:
Could we get a `deprecated()` call in here?

Reply:
I'd be hesitant to add a call to `deprecated` there, until we actually took steps to remove it from the code, which should probably be a separate PR. There's currently a <span>@</span>deprecated tag on the function as a hint to the IDE.